`crypto-refresh`: add support for Argon2 S2K

larabr opened the pull request:
Added support for encrypting and decrypting messages or keys that use Argon2 as S2K.
This feature is backwards compatible, no breaking changes.

Newly introduced config fields (relevant for encryption only):
- `config.s2kType` (defaulting to `enums.s2k.iterated`): s2k to use on password-based encryption as well as private key encryption;
- `config.s2kArgon2Params` (defaulting to "uniformly safe settings" from Argon RFC): parameters to use on encryption when `config.s2kType` is set to `enums.s2k.argon2`;

More technical details:
Argon2 hashes are computed in Wasm, using a module that is only loaded if needed. The Wasm module is automatically reloaded -- to force memory deallocation -- if more than 1GB of memory is used.

larabr committed:
Add support for Argon2 S2K

larabr committed:
Only allow argon2 or iterated s2k as `config.s2kType`

larabr committed:
TS: update Config definition with new s2k fields

larabr committed:
CI: allow argon2 memory-heavy test to fail in browsers

larabr committed:
Add newS2KFromConfig

larabr committed:
Rollup: use platform-specific config and always inline for browser

twiss commented on rollup.config.js:
```diff
@@ -12,6 +12,10 @@ import { wasm } from '@rollup/plugin-wasm';
 const nodeDependencies = Object.keys(pkg.dependencies);
+const wasmOptions = {
+  node: { targetEnv: 'node' },
+  browser: { targetEnv: 'browser', maxFileSize: Number.MAX_SAFE_INTEGER } // always inlline (our wasm files are small)
```
`maxFileSize` can just be `null` or `undefined` to disable the check entirely

larabr committed:
Rename argon2id chunk to include package name

larabr committed:
Simplify chunk renaming

If a package results in two chunks with the same name, then we should consider changing the build to create a single chunk for the given module
